File: benchmark/bench.js
// bnech.js  Run inside of NodeJS

var fs = require("fs");

function capitalize(string) {
  return string.charAt(0).toUpperCase() + string.slice(1);
}

function processLine(line) {
  var words = line.split(" ");
  for (var i = 0; i < words.length; i++) {
    words[i] = capitalize(words[i]);
  }

  return words.join(" ");
}

function run() {
  var data = fs.readFileSync("textfile_1000kb.txt", "utf-8");
  var lines = data.split("\n");

  for (var i = 0; i < lines.length; i++) {
    lines[i] = processLine(lines[i]);
  }

  var cap_lines = lines.join("\n");

  fs.writeFile("textfile_caps_nodejs.txt", cap_lines, function(err) {
    if (err) {
      return console.log(err);
    }
  });

  return cap_lines;
}

console.log(run());

File: benchmark/bench.py
def process(data):
    lines = data.split('\n')
    return "\n".join([process_line(line) for line in lines])


def process_line(line):
    words = line.split(' ')
    new = [w.capitalize() for w in words]
    return " ".join(new)


if __name__ == '__main__':
    data = open('textfile_1000kb.txt').read()
    capitalzed= process(data)
    print capitalzed
    filew=open("textfile_caps_python.txt","w") 
    filew.writelines(capitalzed)
    filew.close()